core/payment-driver/base/src/bus.rs

```diff
@@ -74,6 +74,9 @@ pub async fn bind_service<Driver: PaymentDriver + 'static>(
         .bind_with_processor(
             move |_, dr, c, m| async move { dr.sign_payment( c, m).await }
         )
+        .bind_with_processor(
+            move |_, dr, c, m| async move { dr.sign_payment_canonicalized(c, m).await }
+        )
```

Comment:
I would reverse naming in driver.
`sign_payment` is our basic version so it should do the canonicalization.
And we need second function `sign_legacy` or maybe some better name.